<launch>
  <arg name="namespace" default="robot1" />
  <arg name="verbose" default="true" />
  <arg name="model_type" default="swin" />
  <arg name="enable_pubsub" default="true" />
  <arg name="out_debug_img" default="false" />                                                                         
  <arg name="out_debug_segimg" default="false" />
  <arg name="confidence_threshold" default="0.5" />
  <arg name="input_image" default="rgb_left" />
  <arg name="input_depth" default="depth_left" />
  <arg name="input_camera_info" default="camera_info_left" />
  <arg name="compressed" default="true" />
  <arg name="device" default="auto" />
  <arg name="vocabulary" default="lvis" />
  <arg name="custom_vocabulary" default="" />
  <arg name="output_highest" default="false" />
  <param name="use_sim_time" value="true" />

  <group ns='$(arg namespace)'>
    <node name="detic_segmentor" pkg="detic_ros" type="node.py" output="screen" >
      <remap from="~input_image" to="$(arg input_image)"/>
      <remap from="~input_depth" to="$(arg input_depth)"/>
      <remap from="~input_camera_info" to="$(arg input_camera_info)"/>
      <param name="verbose" value="$(arg verbose)"/>
      <param name="use_jsk_msgs" value="false"/>
      <param name="model_type" value="$(arg model_type)" />
      <param name="enable_pubsub" value="$(arg enable_pubsub)"/>
      <param name="out_debug_img" value="$(arg out_debug_img)"/>
      <param name="out_debug_segimg" value="$(arg out_debug_segimg)"/>
      <param name="confidence_threshold" value="$(arg confidence_threshold)"/>
      <param name="device" value="$(arg device)"/>
      <param name="vocabulary" value="$(arg vocabulary)" />
      <param name="custom_vocabulary" value="$(arg custom_vocabulary)" />
      <param name="output_highest" value="$(arg output_highest)" />
    </node>
  </group>

</launch>